app/models/authenticator/httpd.rb

```diff
@@ -123,6 +123,7 @@ def username_to_upn_name(user_attrs)
       return user_attrs[:username] if user_attrs[:domain].nil?
 
       user_name = user_attrs[:username].split("@").first
+      user_name = user_name.split("\\").last if user_name.include?("\\")
```

Comment:
found my issue.

The `user_name` is the short name.
It is smart enough to shorten the unix-ey way of displaying the user: `username@domain` but is not smart enough to shorten the AD/windows name: `domain\user`.

Other areas of the code do split it up correctly.
So this is good for me

Reply:
I'm not sure how to test this but yes, it looks like the right change